pilopt/src/inliner.rs

```diff
@@ -11,46 +14,48 @@ pub fn replace_constrained_witness_columns<
     T: FieldElement,
     V: Ord + Clone + Hash + Eq + Display,
 >(
-    mut constraint_system: ConstraintSystem<T, V>,
+    constraint_system: ConstraintSystem<T, V>,
     max_degree: usize,
 ) -> ConstraintSystem<T, V> {
-    loop {
-        if !try_apply_substitution(&mut constraint_system, max_degree) {
```

Comment:
Why can we do this without this loop now? Before, it would re-try searching for a variable from the beginning. Is it obvious that we would not find another variable until the first substitution point? Should we at least re-consider the constraint we just substituted in?